linode_api4/linode_client.py

```diff
@@ -1618,6 +1619,69 @@ def image_create(self, disk, label=None, description=None):
 
         return Image(self, result['id'], result)
 
+    def image_create_upload(self, label: str, region: str, description: str=None) -> Tuple[Image, str]:
+        """
+        Creates a new Image and returns the corresponding upload URL.
+        https://www.linode.com/docs/api/images/#image-upload
+
+        :param label: The label of the Image to create.
+        :type label: str
+        :param region: The region to upload to. Once the image has been created, it can be used in any region.
+        :type region: str
+        :param description: The description for the new Image.
+        :type description: str
+
+        :returns: A tuple containing the new image and the image upload URL.
+        :rtype: (Image, str)
+        """
+        params = {
+            "label": label,
+            "region": region
+        }
+
+        if description is not None:
+            params["description"] = description
+
+        result = self.post("/images/upload", data=params)
+
+        if "image" not in result:
+            raise UnexpectedResponseError('Unexpected response when creating an '
```

Comment:
Is it possible to throw a more detailed error message? i.e. the reason why image isn't in the result.

Reply:
I think errors returned by the API should be returned during the POST call, so this check _should_ only occur if the API returns a weird response, etc. I do wonder if these checks are necessary in that case.